src/AppBundle/Admin/AdministratorAdmin.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace AppBundle\Admin;
+
+use AppBundle\Entity\Administrator;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+
+class AdministratorAdmin extends AbstractAdmin
+{
+    /**
+     * @var EncoderFactoryInterface
+     */
+    private $encoders;
+
+    /**
+     * @param Administrator $admin
+     */
+    public function prePersist($admin)
+    {
+        parent::prePersist($admin);
+
+        $admin->setPassword($this->encoders->getEncoder($admin)->encodePassword($admin->plainPassword, null));
```

Comment:
Ok, I  understand. Still, having this public property being changed is enough to persist a new encoded password, it is not very safe, right?

The best way IMHO is to use a model transformer instead of both listener:
```php
protected function configureFormFields(FormMapper $formMapper)
    {
        $isCreation = null === $this->getSubject()->getId();

        $formMapper
            // ...
            ->add(
                // Use the encoded property for mapping, even get rid of the plain one
                $this->create('password', RepeatedType::class, [
                    'first_options' => [
                        'label' => 'Mot de passe',
                    ],
                    'second_options' => [
                        'label' => 'Confirmation',
                    ],
                    'type' => PasswordType::class,
                    'required' => $isCreation,
                ])
                ->addModelTransformer(new CallbackTransformer(
                    function ($encoded) {
                        return ''; // or ****
                    },
                    function ($plain) {
                        return is_string($plain) && '' !== $plain
                            ? $this->encoders->getEncoder($admin)->encodePassword($admin->plainPassword, null)
                            : null
                    ;
            )
    ;
}